include/swift/Reflection/ReflectionContext.h

```diff
@@ -378,30 +378,39 @@ class ReflectionContext
     return readPECOFFSections(ImageStart);
   }
 
-  template <typename T> bool readELFSections(RemoteAddress ImageStart) {
+  template <typename T>
+  bool readELFSections(RemoteAddress ImageStart, const void *FileBuffer,
+                       uint64_t FileSize) {
     auto Buf =
         this->getReader().readBytes(ImageStart, sizeof(typename T::Header));
 
     auto Hdr = reinterpret_cast<const typename T::Header *>(Buf.get());
     assert(Hdr->getFileClass() == T::ELFClass && "invalid ELF file class");
 
-    // From the header, grab informations about the section header table.
-    auto SectionHdrAddress = ImageStart.getAddressData() + Hdr->e_shoff;
-    auto SectionHdrNumEntries = Hdr->e_shnum;
-    auto SectionEntrySize = Hdr->e_shentsize;
+    uint64_t FileBufferStartAddress = (uint64_t)FileBuffer;
```

Comment:
Are we checking for `!FileBuffer` somewhere?

Reply:
You're right, I'll add that check.